src/coreclr/jit/emit.cpp

```diff
@@ -7273,6 +7273,90 @@ CORINFO_FIELD_HANDLE emitter::emitFltOrDblConst(double constValue, emitAttr attr
     return emitComp->eeFindJitDataOffs(cnum);
 }
 
+//------------------------------------------------------------------------
+// emitSimd8Const: Create a simd8 data section constant.
+//
+// Arguments:
+//    constValue - constant value
+//
+// Return Value:
+//    A field handle representing the data offset to access the constant.
+//
+CORINFO_FIELD_HANDLE emitter::emitSimd8Const(simd8_t constValue)
```

Comment:
It seems like a lot of code duplication, I wonder if you could collapse these 3 functions into one with a pointer to the data (`simd_t*` or even just `void*` and a `var_types` argument).

Reply:
There is additional code in any way you do this. This at least makes it less error prone overall.

-- The duplication is either here or at the call site to ensure the right data is passed. These functions likewise get "more complex" if we take a type and data pointer.